src/ddmd/parse.d

```diff
@@ -7051,12 +7147,43 @@ final class Parser : Lexer
                 break;
             }
         case TOKdollar:
-            if (!inBrackets)
-                error("'$' is valid only inside [] of index or slice");
-            e = new DollarExp(loc);
+            Token* t1 = peek(&token);
+            if (t1.value == TOKstring)
+            {
+                OutBuffer result;
+                int codeSectionCount = interp(t1.ustring[0..t1.len], result);
+                const(char)[] stringExpStr;
+                if (codeSectionCount == 0)
+                {
+                    stringExpStr = t1.ustring[0..t1.len];
+                }
+                else if (codeSectionCount < 0)
+                {
+                    nextToken();
+                    error(result.peekString());
+                    mem.xfree(result.extractData());
+                    goto Lerr;
+                }
+                else
+                {
+                    stringExpStr = result.peekSlice();
+                }
+
+                auto s = cast(char*)mem.xmalloc(stringExpStr.length);
```

Comment:
use allocmemory here, not the nasty xmalloc which costs atlot.
and is only for temporary strings which are supposed to be freed.
please just alloc the buffer once and reuse it if possible